### PR TITLE
GP-47 Migrate lambda-math-functions

### PR DESCRIPTION
**Pull request by MK-JavaDeveloper**
I'm not sure if I did everything right.

Did I have to sort @Order in order in the test?

### CODE REVIEW

**Comment by mathodcoast** [comment]
We added test ordering to provide a natural exercises completion. It should help to do exercises in the most right order.

**Comment by mathodcoast** [comment]
Your ordering seems right, but you can also make corresponding methods sequence in the test class.

File: 5-0-functional-programming/src/test/java/com/bobocode/lambda_math_functions/FunctionsTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        Function<Integer, Integer> signFunction = integerFunctionMap.getFunction("sgn");
+
+        int actualResult = signFunction.apply(0);
+
+        assertEquals(0, actualResult);
+    }
+
+    @Test
+    @Order(8)
+    void getGetUnknownFunction() {
```
Comment (mathodcoast) [inline 5-0-functional-programming/src/test/java/com/bobocode/lambda_math_functions/FunctionsTest.java:93]: I think we have redundant appendix here.

File: 5-0-functional-programming/src/test/java/com/bobocode/lambda_math_functions/FunctionsTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    @Test
+    @Order(7)
+    void squareFunction() {
+        Function<Integer, Integer> squareFunction = integerFunctionMap.getFunction("square");
+
+        int actualResult = squareFunction.apply(5);
+
+        assertEquals(25, actualResult);
```
Comment (mathodcoast) [inline 5-0-functional-programming/src/test/java/com/bobocode/lambda_math_functions/FunctionsTest.java:28]: You should replace all the assertions by "AssertJ" corresponding to [Migration Guide](https://bobocode.atlassian.net/wiki/spaces/UJE/pages/406290441/Migration+guide) .

**Review by mathodcoast — CHANGES_REQUESTED** [review]

File: 5-0-functional-programming/src/test/java/com/bobocode/lambda_math_functions/FunctionsTest.java (hunk 1 above)
Comment (MK-JavaDeveloper) [reply 5-0-functional-programming/src/test/java/com/bobocode/lambda_math_functions/FunctionsTest.java]: Sorry, I have already corrected it.

File: 5-0-functional-programming/src/test/java/com/bobocode/lambda_math_functions/FunctionsTest.java (hunk 2 above)
Comment (MK-JavaDeveloper) [reply 5-0-functional-programming/src/test/java/com/bobocode/lambda_math_functions/FunctionsTest.java]: I corrected it, at least I think so. 
I do not fully understand what I am doing and how to do it correctly.
But I hope this time is what it was about it.

File: 5-0-functional-programming/src/test/java/com/bobocode/lambda_math_functions/FunctionsTest.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
+package com.bobocode.lambda_math_functions;
+
+import com.bobocode.InvalidFunctionNameException;
+import org.junit.jupiter.api.*;
+
+import java.util.function.Function;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Comment (mathodcoast) [inline 5-0-functional-programming/src/test/java/com/bobocode/lambda_math_functions/FunctionsTest.java:8]: If everything is correct these imports are redundant.

File: 5-0-functional-programming/src/test/java/com/bobocode/lambda_math_functions/FunctionsTest.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+        int actualResult = signFunction.apply(0);
+
+        assertThat(result).isEqualTo(0);
+    }
+
+    @Test
+    @Order(8)
+    void getUnknownFunction() {
+        assertThrows(InvalidFunctionNameException.class, () -> integerFunctionMap.getFunction("sqrt"));
```
Comment (mathodcoast) [inline 5-0-functional-programming/src/test/java/com/bobocode/lambda_math_functions/FunctionsTest.java:94]: AssertJ also supports exceptions. [AssertJ Doc](https://assertj.github.io/doc/#assertj-core-exception-assertions)

**Review by mathodcoast — CHANGES_REQUESTED** [review]

File: 5-0-functional-programming/src/test/java/com/bobocode/lambda_math_functions/FunctionsTest.java (hunk 4 above)
Comment (MK-JavaDeveloper) [reply 5-0-functional-programming/src/test/java/com/bobocode/lambda_math_functions/FunctionsTest.java:94]: Sorry, but unfortunately I don't know how to change it. Could you please tell me how to do this. What exactly needs to be written. Unfortunately, I don't understand it yet.

File: 5-0-functional-programming/src/test/java/com/bobocode/lambda_math_functions/FunctionsTest.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
+package com.bobocode.lambda_math_functions;
+
```
Comment (mathodcoast) [inline 5-0-functional-programming/src/test/java/com/bobocode/lambda_math_functions/FunctionsTest.java:2]: When I wrote about redundant imports I meant that we don't need imports of old assertions (`assertEquals`, `assertThrows`). But we can't perform this class without others necessary imports.

**Review by mathodcoast — CHANGES_REQUESTED** [review]

**Review by mathodcoast — APPROVED** [review]

**Pull request merged**